### PR TITLE
[SYSTEMDS-310] Python API Refactor

### PR DESCRIPTION
**Pull request by Baunsgaard**
Read comment bellow

### CODE REVIEW

File: src/main/java/org/apache/sysds/api/PythonDMLScript.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			GatewayServer GwS = new GatewayServer(new PythonDMLScript(), Integer.parseInt(port));
+			// GwS is non blocking, if first argument is true then it becomes blocking,
+			// but then the python process will not know when the JVM instance is ready.
+			GwS.start();
+			// message the python interface that the JVM is ready.
+			System.out.println("GatewayServer Started");
+
+			// Hack block this process until python stops the gateway.
+			// https://stackoverflow.com/questions/30845546/how-can-we-make-a-thread-to-sleep-for-a-infinite-time-in-java#30845606
+			new CyclicBarrier(2).await();
```
[the remark this answers is not in the document]
Comment (mboehm7) [reply src/main/java/org/apache/sysds/api/PythonDMLScript.java:58]: Maybe we could add a `GatewayServerListener` which in its `serverPostShutdown` simply calls a `notify` on a monitor we `wait` on here?

[the remark this answers is not in the document]
Comment (Baunsgaard) [reply src/main/java/org/apache/sysds/api/PythonDMLScript.java]: Now changed to use that, thanks for the suggestion, absolutely the right call.

**Comment by Baunsgaard** [comment]
Maybe a new committer want to review this. @kev-inn ?

**Comment by mboehm7** [comment]
LGTM - that's a very nice patch @Baunsgaard. 

Just some minor comments: (1) as I've seen it before: the correct spelling is built-in functions/algorithms (not build-in), (2) usually we try to delete commented code (e.g., `runFedTest.sh`) and avoid references to transient links as they might change (people will be able to find the quickstart guide of py4j). I fixed (1) but left (2) for now.

**Pull request closed without merging**